lib/spack/spack/util/executable.py

```diff
@@ -120,6 +124,29 @@ def __call__(self, *args, **kwargs):
         By default, the subprocess inherits the parent's file descriptors.
 
         """
+
+        def process_cmd_output():
```

Comment:
Python functions can access identifiers declared in their parent scope prior to the identifier declaration/definition since identifier references are evaluated at invocation time. If we were trying to assign something to `output` here, that would be a different story (we'd have to use the `nonlocal` keyword)

Reply:
```
>>> def write_name():
...   print(name)
...
>>> write_name()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 2, in write_name
NameError: name 'name' is not defined
>>> name = "frank"
>>> write_name()
frank
>>>
```